src/metabase/query_processor/util/nest_query.clj

```diff
@@ -134,9 +148,11 @@
     :expression
     (raise-source-query-expression-ref query &match)
 
-    ;; mark all Fields at the new top level as `:qp/ignore-coercion` so QP implementations know not to apply coercion or
-    ;; whatever to them a second time.
-    [:field _id-or-name (_opts :guard (every-pred :temporal-unit (complement :qp/ignore-coercion)))]
+    ;; Mark all Fields at the new top level as `:qp/ignore-coercion` so QP implementations know not to apply coercion
+    ;; or whatever to them a second time.
+    ;; In fact, we don't mark all Fields, only the ones we deem coercible. Marking all would make a bunch of tests
+    ;; fail, but it might still make sense. For example, #48721 would have been avoided by unconditional marking.
```

Comment:
@camsaul, would it be a bad idea to really add `:qp/ignore-coercion` to all field refs?

Reply:
I think this is fine for now. This will get ported to MBQL lib and new refs in the middle term (< 6 months) and there are better ways to address this.